### PR TITLE
fix(runtime): `Deno.memoryUsage().rss` should return correct value

### LINKED ISSUE #16581: Deno.memoryUsage Returning rss < heapTotal
Installed version of deno: 1.27.2. This is an issue across multiple applications and not specific to any particular tech stack.

I'm currently having an issue where the rss returned from Deno.memoryUsage (across applications) is returning an rss size that is less than the heapTotal. This is an issue that just started in the past few days. It appears as though heapTotal is acting like the rss and the rss is acting like the heapTotal; rss is shrinking in tandem with the heapUsed as garbage collection occurs, but the heapTotal is consistently the largest variable. Rss starts higher in some applications and then crosses below heapTotal in applications without known memory leaks. Below is an example of console logs from an application with a known memory leak. 

Screen Shot 2022-11-09 at 3 21 17 PM

Below is an example of the memory for an application that has no known memory leaks graphed over 10 seconds. Notice the rss starts larger and then crosses over the heapTotal.

Screen Shot 2022-11-09 at 3 24 52 PM

This is a ten second snapshot of the same application, about a minute later. Notice that the rss has stayed below the heapTotal. 

Screen Shot 2022-11-09 at 3 26 25 PM

Here is an example of console logs in this application of Deno.memoryUsage. This is a direct invocation, showing that the graph of the memory is not the issue, but the function itself is giving back data with the rss < heapTotal. 

Screen Shot 2022-11-09 at 3 27 34 PM

Could someone provide some insight onto what's going on? My understanding based on every article I've read and from the Deno docs is that rss should represent the total memory of the process in RAM. Which means it should encompass both heapTotal, heapUsed, and external memory. It should never fall below heapTotal. 

image

Screen Shot 2022-11-09 at 3 30 22 PM

### PR DESCRIPTION
**Pull request by dsherret**
Closes #16581

### CODE REVIEW

**Comment by bartlomieju** [comment]
@kangelopoulos hopefully this fixes your problem, here's example numbers I get on mac:
```
$ cargo run
Deno 1.29.1
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> Deno.memoryUsage()
{ rss: 203079680, heapTotal: 3964928, heapUsed: 3556760, external: 225 }
>
```

File: runtime/ops/process.rs [hunk 1]
```diff
@@ -360,3 +366,124 @@ fn op_kill(
...
+    let current_process = GetCurrentProcess();
+    let mut pmc: PROCESS_MEMORY_COUNTERS = std::mem::zeroed();
+
+    if GetProcessMemoryInfo(
+      current_process,
+      &mut pmc,
+      std::mem::size_of::<PROCESS_MEMORY_COUNTERS>() as DWORD,
+    ) != FALSE
+    {
+      pmc.WorkingSetSize
```
Comment (dsherret) [inline runtime/ops/process.rs:484]: Verified this returns back about the same number as when I do:

```
> wmic process where processid=<pid goes here> get WorkingSetSize
```

Comment (bartlomieju) [reply runtime/ops/process.rs:484]: Verified the same on macos

**Comment by magurotuna** [comment]
on Ubuntu 22.04.1

```
$ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.23s
     Running `target/debug/deno`
Deno 1.29.1
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> Deno.memoryUsage()
{ rss: 217903104, heapTotal: 3915776, heapUsed: 3585416, external: 225 }
```

For comparison, here's the result of when running on Deno 1.29.1 (released one):

```
$ deno
Deno 1.29.1
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> Deno.memoryUsage()
{ rss: 3899392, heapTotal: 3907584, heapUsed: 3576700, external: 225 }
```

**Comment by bartlomieju** [comment]
Thanks for checking @magurotuna!

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**